client/src/main/java/org/apache/predictionio/sdk/java/EngineClient.java

```diff
@@ -106,7 +108,6 @@ public FutureAPIResponse sendQueryAsFuture(Map<String, Object> query)
     String requestJsonString = gson.toJson(query);
     builder.setBody(requestJsonString);
     builder.setHeader("Content-Type", "application/json");
-    builder.setHeader("Content-Length", "" + requestJsonString.length());
```

Comment:
Removing this line seems to fix the issue - I haven't found anything that relies on the Content-Length being explicitly set here, so hopefully it won't cause other problems.

Reply:
Bob, I am not sure if I like this, looks like a workaround. The issue could due to the fact that the response is sent as bytes, and the Java string length may not match the bytes length. Perhaps, try being explicit with encoding:
Content-type: application/json; charset=utf-8
And then your content length would be requestJsonString.getBytes("UTF-8").length

Reply:
Looks like using getBytes works, so I made that change and pushed it up here - thanks